src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs

```diff
@@ -0,0 +1,405 @@
+        stderr: Option<String>,
+        terminal: bool,
+        oci_process: OCIProcess,
+    ) -> Result<()> {
+        let process = Process::new(
+            container_process,
+            self.pid,
+            &self.config.bundle,
+            stdin,
+            stdout,
+            stderr,
+            terminal,
+        );
+        let exec = Exec {
+            process,
+            oci_process,
+        };
+        let mut inner = self.inner.write().await;
+        inner.add_exec_process(&container_process.exec_id, exec);
+        Ok(())
+    }
+
+    pub async fn close_io(&self, container_process: &ContainerProcess) -> Result<()> {
+        let mut inner = self.inner.write().await;
+        inner.close_io(container_process).await
+    }
+
+    pub async fn stop_process(&self, container_process: &ContainerProcess) -> Result<()> {
+        let mut inner = self.inner.write().await;
+        inner
+            .stop_process(container_process, true)
+            .await
+            .context("stop process")
+    }
+
+    pub async fn pause(&self) -> Result<()> {
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Paused {
+            warn!(self.logger, "container is paused no need to pause");
+            return Ok(());
+        }
+        self.agent
+            .pause_container(self.container_id.clone().into())
+            .await
+            .context("agent pause container")?;
+        Ok(())
+    }
+
+    pub async fn resume(&self) -> Result<()> {
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Running {
+            warn!(self.logger, "container is running no need to resume");
+            return Ok(());
+        }
+        self.agent
+            .resume_container(self.container_id.clone().into())
+            .await
+            .context("agent pause container")?;
+        Ok(())
+    }
+
+    pub async fn resize_pty(
+        &self,
+        process: &ContainerProcess,
+        width: u32,
+        height: u32,
+    ) -> Result<()> {
+        let logger = logger_with_process(process);
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Running {
+            warn!(logger, "container is running no need to resume");
+            return Ok(());
+        }
+        self.agent
+            .tty_win_resize(agent::TtyWinResizeRequest {
+                process_id: process.clone().into(),
+                row: height,
+                column: width,
+            })
+            .await
+            .context("resize pty")?;
+        Ok(())
+    }
+
+    pub async fn stats(&self) -> Result<Option<agent::StatsContainerResponse>> {
+        let stats_resp = self
+            .agent
+            .stats_container(self.container_id.clone().into())
+            .await
+            .context("agent stats container")?;
+        Ok(Some(stats_resp))
+    }
+
+    // to do
```

Comment:
`to do` what？

Reply:
Thanks for your review, has been removed.